server/lib/auth.js

```diff
@@ -1,116 +1,138 @@
+ */
 const setUserAuthCookie = async (res, user) => {
-    const userSession = await generateUserSession(user)
+    const userSession = await generateUserSession(user); // Generate new user session
+    const expiry = new Date(Date.now() + 1000 * 60 * 60 * 12); // Expire cookie in 12hrs
+    
+    // Set two cookies: one for the session token and another for the CSRF token
     res.setHeader("Set-Cookie", [
         serialize("_myclassroom_session", generateUserAuthToken(user), {
             path: "/",
-            httpOnly: true, //this means that the cookie is only accessible to the server and sent along in reequests. The client side has no direct access to it
-            expires: new Date(Date.now() + 1000 * 60 * 60 * 12) //expires in 12 hours
+            httpOnly: true, // Cookie is HTTP only for security
+            expires: expiry // Set expiry for cookie
         }),
-        serialize("XSRF-TOKEN", userSession.csrfToken, { //protect from CSRF by giving the client a token that can be accessed and attached in request headers for checking
+        serialize("xsrf-token", userSession.csrfToken, {
             path: "/",
-            expires: new Date(Date.now() + 1000 * 60 * 60 * 12) //expires in 12 hours
+            expires: expiry // Set expiry for CSRF token
         })
-    ])
-}
-exports.setUserAuthCookie = setUserAuthCookie
+    ]);
+};
 
+/**
+ * Removes the authentication cookie from the user's browser
+ * @param {Object} req - The request object to read cookies from
+ * @param {Object} res - The response object to set cookie headers on
+ */
 const removeUserAuthCookie = async (req, res) => {
-    await db.Session.update({ expires: moment().utc()}, { where: { csrfToken: extractCsrfToken(req) }})
+    // Update session to expire immediately
+    await db.Session.update({ expires: moment().utc()}, { where: { csrfToken: req.cookies["xsrf-token"]}});
+    
+    // Clear both cookies by setting them to empty strings
     res.setHeader("Set-Cookie", [
-        serialize("_myclassroom_session", "", {
-            path: "/",
-            httpOnly: true
-        }),
-        serialize("XSRF-TOKEN", "", {
-            path: "/"
-        })
-    ])
-}
-exports.removeUserAuthCookie = removeUserAuthCookie
+        serialize("_myclassroom_session", "", { path: "/", httpOnly: true }),
+        serialize("xsrf-token", "", { path: "/" })
+    ]);
+};
 
-async function requireAuthentication(req, res, next) {
-    const token = extractToken(req)
-    const csrfToken = extractCsrfToken(req)
+/**
+ * Middleware to require authentication for a route
+ * @param {Object} req - The request object
+ * @param {Object} res - The response object
+ * @param {Function} next - The next middleware function in the stack
+ */
+const requireAuthentication = async (req, res, next) => {
+    // Extract session/CSRF token from cookies
+    const token = req.cookies["_myclassroom_session"];
+//    const csrfToken = req.headers["x-xsrf-token"];
```

Comment:
Might as well remove this commented code

Reply:
Added in the latest commit